psamm/datasource/native.py

```diff
@@ -174,28 +174,39 @@ class NativeModel(object):
     directory subtree that specifies part of the model.
     """
 
-    def __init__(self, path):
-        """Create a model from the specified model file or directory"""
-
-        if os.path.isfile(path):
-            self._context = FilePathContext(path)
-            with open(self._context.filepath, 'r') as f:
-                self._model = yaml_load(f)
-        else:
-            # Try to open the default file
-            for filename in DEFAULT_MODEL:
-                try:
-                    self._context = FilePathContext(
-                        os.path.join(path, filename))
-                    with open(self._context.filepath, 'r') as f:
-                        self._model = yaml_load(f)
-                        break
-                except:
-                    logger.debug('Failed to load model file', exc_info=True)
+    def __init__(self, model_from):
+        """Create a model from the specified model file, directory,
+            long string, open file, or dictionary"""
+        if isinstance(model_from, string_types):
+            if 'name: ' in model_from:
```

Comment:
This is clever but still not safe... The `name:` may not appear in the string even though it is a model. I don't know if there is really a good solution to allowing a string to work as either an inline YAML or a path to a YAML file. It may be necessary to split off the file loading to (perhaps) a factory method?